security-analysis/security-analysis-api/src/main/java/com/powsybl/security/distributed/DistributedSecurityAnalysis.java

```diff
@@ -89,12 +107,51 @@ protected SecurityAnalysisResult readResults(Path workingDir) {
          */
         @Override
         protected List<CommandExecution> buildCommandExecution() {
-            SimpleCommand cmd = baseCommand("security-analysis-task")
+            return Collections.singletonList(new CommandExecution(baseCmdBuilder().build(), actualTaskCount, 1));
+        }
+
+        SimpleCommandBuilder baseCmdBuilder() {
+            return baseCommand(SA_TASK_CMD_ID)
                     .option(TASK, i -> new Partition(i + 1, actualTaskCount).toString())
-                    .option("output-file", this::getOutputFileName)
-                    .option("output-format", "JSON")
+                    .option(OUTPUT_FILE_OPTION, this::getOutputFileName)
+                    .option(OUTPUT_FORMAT_OPTION, "JSON");
+        }
+
+    }
+
+    private class SubTaskWithLogHandler extends AbstractExecutionHandler<SecurityAnalysisResultWithLog> {
+
+        private SubTaskHandler simpleSubTaskHandler;
+
+        public SubTaskWithLogHandler(SubTaskHandler subTaskHandler) {
+            this.simpleSubTaskHandler = Objects.requireNonNull(subTaskHandler);
+        }
+
+        @Override
+        public List<CommandExecution> before(Path workingDir) throws IOException {
+            simpleSubTaskHandler.before(workingDir); // invocation just for prepare input files
+            // override cmd with log
+            SimpleCommand cmd = simpleSubTaskHandler.baseCmdBuilder()
+                    .option(OUTPUT_LOG_OPTION, this::getLogOutputName)
                     .build();
-            return Collections.singletonList(new CommandExecution(cmd, actualTaskCount, 1));
+            return Collections.singletonList(new CommandExecution(cmd, simpleSubTaskHandler.actualTaskCount, 1));
+        }
+
+        private String getLogOutputName(int taskNumber) {
+            return "logs_" + taskNumber + ".xml.gz";
```

Comment:
After yesterday discussion and some testing, I must say I think it would be better to have a flat zip file in the end (a zip containing unzipped files).
For a user it will be far more handy. If we don't do this, as a client application, I will need to implement that inside my application to give something usable by an end user (will need to untar, gunzip, gunzip individual files, put them again in a zip).

Reply:
PS:
a possibility is to have an option, but it may not be very useful.